release notes for stable-2.10.1 (#6032)
linkerd/linkerd2: CHANGES.md

# Changes

## stable-2.10.1

This stable release adds CLI support for Apple Silicon M1 chips and support for
SMI's TrafficSplit `v1alpha2`.

There are several proxy fixes: handling `FailedPrecondition` errors gracefully,
inbound TLS detection from non-meshed workloads, and using the correct cached

> client when the proxy is in ingress moode. The logging infrastructure has also

Reviewer: ```suggestion
client when the proxy is in ingress mode. The logging infrastructure has also
```

been improved to reduce memory pressure in high-connection environments.

On the control-plane side, there have been several improvements to the
destination service such as support for Host IP lookups and ignoring pods

> "Terminating State". It also updates the proxy-injector to add opaque ports

Reviewer: ```suggestion
in "Terminating" state. It also updates the proxy-injector to add opaque ports
```

annotation to pods if their namespace has it set.

On the CLI side, `linkerd repair` has been updated to be aware about the control-plane

> version and suggest the relevant suggestion to generate the right config. Various

Reviewer: ```suggestion
version and suggest the relevant version to generate the right config. Various
```

    that already had Linkerd installed
  * Added a loading spinner to the linkerd check command when running
    extension checks
  * Added installNamespace toggle in the jaeger extension's install.
    (thanks @jijeesh!)
  * Updated healthcheck pkg to have hintBaseURL configurable, useful
    for external extensions using that pkg
  * Fixed TCP read and write bytes/sec calculations to group by label
    based off inbound or outbound traffic
  * Fixed an issue in linkerd inject where the wrong annotation would
    be added when using --ingress flag
  * Updated `linkerd repair` to be aware of the client and server versions

* Helm:
  * Aligned the Helm installation heartbeat schedule to match that of the CLI

* Viz:
  * Fixed an issue where the topology graph in the dashboard was no
    longer draggable.
  * Updated dashboard build to use webpack v5
  * Added CA certs to the Viz extension's metrics-api container so
    that it can validate the certifcate of an external Prometheus
  * Removed components from the control plane dashboard that now
    are part of the Viz extension
  * Changed web's base image from debian to scratch

* Multicluster:

>   * Fixed an issue with Multicluster's serivce mirror where it's endpoint

Reviewer: ```suggestion
  * Fixed an issue with Multicluster's service mirror where its endpoint
 ```
